elytron/src/test/java/org/wildfly/extension/elytron/RealmsTestCase.java

```diff
@@ -727,6 +728,46 @@ public void testPathToJAASConfigFileDoesNotExist() throws Exception {
         Assert.assertTrue(response.asString().contains("JAAS configuration file does not exist."));
     }
 
+    @Test
+    public void testDistributedRealm() throws Exception {
+        KernelServices services = super.createKernelServicesBuilder(new TestEnvironment()).setSubsystemXmlResource("realms-test.xml").build();
+        if (!services.isSuccessfulBoot()) {
+            Assert.fail(services.getBootError().toString());
+        }
+
+        ServiceName serviceName = Capabilities.SECURITY_REALM_RUNTIME_CAPABILITY.getCapabilityServiceName("DistributedRealmOne");
+        SecurityRealm distributedRealm = (SecurityRealm) services.getContainer().getService(serviceName).getValue();
+        testDistributedRealmSuccessful(distributedRealm);
+
+        ServiceName serviceName2 = Capabilities.SECURITY_REALM_RUNTIME_CAPABILITY.getCapabilityServiceName("DistributedRealmTwo");
+        SecurityRealm distributedRealm2 = (SecurityRealm) services.getContainer().getService(serviceName2).getValue();
+        testDistributedRealmFailure(distributedRealm2);
+
+        ServiceName serviceName3 = Capabilities.SECURITY_REALM_RUNTIME_CAPABILITY.getCapabilityServiceName("DistributedRealmThree");
+        SecurityRealm distributedRealm3 = (SecurityRealm) services.getContainer().getService(serviceName3).getValue();
+        testDistributedRealmSuccessful(distributedRealm3);
+
+        ServiceName serviceName4 = Capabilities.SECURITY_REALM_RUNTIME_CAPABILITY.getCapabilityServiceName("DistributedRealmFour");
+        SecurityRealm distributedRealm4 = (SecurityRealm) services.getContainer().getService(serviceName4).getValue();
+        testDistributedRealmSuccessful(distributedRealm4);
+    }
+
+    private void testDistributedRealmSuccessful(SecurityRealm distributedRealm) throws RealmUnavailableException {
```

Comment:
Effect of `emit-events` is not tested.

Reply:
@lvydra Just to check, has this comment from @OndrejKotek been addressed yet?

Reply:
@fjuma Yes, this one has already been addressed.